File: Lab3/calculator.java
//Basic user input calculator for Java 
public class Calculator {

	public static <T implements Number> T compute(T in1, T in2, String op) {
			T out;
			if (input[1].equals("+")) out = in1 + in2;
			else if(input[1].equals("-")) out = in1 - in2;
			else if(input[1].equals("*")) out = in1 * in2;
			else if(input[1].equals("/")) out = in1 / in2;
			else out = in1 / in2;
			return out;
	}

	public static void main(String[] args) {
		String[] input = args[0].split(" ");
		String decimal = "\\d*(\\.\\d|\\d\\.)\\d*";
		if (input[0].matches(decimal) || input[2].matches(decimal)){

			double in1 = double.parseDouble(input[0]);
			double in2 = double.parseDouble(input[2]);
			double out = compute(in1, in2, input[1]);

			// if (input[1].equals("+")) out = in1 + in2;
			// else if(input[1].equals("-")) out = in1 - in2;
			// else if(input[1].equals("*")) out = in1 * in2;
			// else if(input[1].equals("/")) out = in1 / in2;
			// else out = in1 / in2;

			System.out.println(args[0] + " = " + out);
		}
		else {
			int in1 = Integer.parseInt(input[0]);
			int in2 = Integer.parseInt(input[2]);
			int out;

			if (input[1].equals("+")) out = in1 + in2;
			else if(input[1].equals("-")) out = in1 - in2;
			else if(input[1].equals("*")) out = in1 * in2;
			else if(input[1].equals("/")) out = in1 / in2;
			else out = in1 / in2;

			System.out.println(args[0] + " = " + out);
	}
}

	
}

File: Lab3/calculator.py
import sys
import re
#takes the equation from the command line argument and splits it up by space
arg = sys.argv[1] 
string = arg.split(' ')
decimal1 = re.findall(r"\d*\.\d*|\d*\.\d*", string[0])
decimal2 = re.findall(r"\d*\.\d*|\d*\.\d*", string[2])

# If arg has no floats then use integer math
if (len(decimal1) == 0) and (len(decimal2) == 0): 
	in1 = int(string[0])
	in2 = int(string[2])
	if string[1] == '+':
		print in1 , '+' , in2, '=', in1 + in2
	elif string[1] == '-':
		print in1 , '-' , in2, '=', in1 - in2
	elif string[1] == '*':
		print in1 , '*' , in2, '=', in1 * in2
	elif string[1] == '/':
		print in1 , '/' , in2, '=', in1 / in2	
# If arg has floats then use decimal math
else:
	in1 = float(string[0])
	in2 = float(string[2])
	if string[1] == '+':
		print in1 , '+' , in2, '=', in1 + in2
	elif string[1] == '-':
		print in1 , '-' , in2, '=', in1 - in2
	elif string[1] == '*':
		print in1 , '*' , in2, '=', in1 * in2
	elif string[1] == '/':
		print in1 , '/' , in2, '=', in1 / in2	





